pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1839,173 +1257,112 @@ func TestServiceRegistryList(t *testing.T) {
 	defer server.Terminate(t)
 
-	svc1 := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}
+	svc1 := svctest.MakeService("foo")
 	ctx := genericapirequest.NewDefaultContext()
-	createdSvc1, _ := storage.Create(ctx, svc1, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	createdService1 := createdSvc1.(*api.Service)
-	if createdService1.Name != "foo" {
-		t.Errorf("Expected foo, but got %v", createdService1.Name)
-	}
-	if !makeIPNet(t).Contains(net.ParseIP(createdService1.Spec.ClusterIPs[0])) {
-		t.Errorf("Unexpected ClusterIP: %s", createdService1.Spec.ClusterIPs[0])
-	}
-
-	svc2 := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "bar"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		}}
+	obj, err := storage.Create(ctx, svc1, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	createdSvc1 := obj.(*api.Service)
+	if createdSvc1.Name != "foo" {
+		t.Errorf("Expected foo, but got %v", createdSvc1.Name)
+	}
+	if !makeIPNet(t).Contains(net.ParseIP(createdSvc1.Spec.ClusterIPs[0])) {
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc1.Spec.ClusterIPs[0])
+	}
+
+	svc2 := svctest.MakeService("bar")
 	ctx = genericapirequest.NewDefaultContext()
-	createdSvc2, _ := storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	createdService2 := createdSvc2.(*api.Service)
-	if createdService2.Name != "bar" {
-		t.Errorf("Expected bar, but got %v", createdService2.Name)
+	obj, err = storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	createdSvc2 := obj.(*api.Service)
+	if createdSvc2.Name != "bar" {
+		t.Errorf("Expected bar, but got %v", createdSvc2.Name)
 	}
-	if !makeIPNet(t).Contains(net.ParseIP(createdService2.Spec.ClusterIPs[0])) {
-		t.Errorf("Unexpected ClusterIP: %s", createdService2.Spec.ClusterIPs[0])
+	if !makeIPNet(t).Contains(net.ParseIP(createdSvc2.Spec.ClusterIPs[0])) {
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc2.Spec.ClusterIPs[0])
 	}
 
 	testIPs := []string{"1.2.3.93", "1.2.3.94", "1.2.3.95", "1.2.3.96"}
-	testIP := ""
+	testIP := "not-an-ip"
 	for _, ip := range testIPs {
 		if !storage.serviceIPAllocatorsByFamily[storage.defaultServiceIPFamily].(*ipallocator.Range).Has(net.ParseIP(ip)) {
 			testIP = ip
 			break
 		}
 	}
```

Comment:
this can be replaced by calling `allocator.Next()`.. then release it before L1303.

Reply:
With all the changes, github has lost track of where this goes, so I have no idea what it means